packages/kit/src/vite/utils.js

```diff
@@ -223,59 +223,84 @@ const find_illegal_rollup_imports = (
 	illegal_imports,
 	seen = new Set()
 ) => {
-	if (seen.has(node.id)) return;
-	seen.add(node.id);
+	const nodeId = normalizePath(node.id);
+	if (seen.has(nodeId)) return null;
+	seen.add(nodeId);
 
-	if (illegal_imports.has(node.id)) {
-		return [{ name: node.id, dynamic }];
+	if (illegal_imports.has(nodeId)) {
+		return [{ name: nodeId, dynamic }];
 	}
 
 	for (const id of node.importedIds) {
 		const child = node_getter(id);
 		const chain =
 			child && find_illegal_rollup_imports(node_getter, child, false, illegal_imports, seen);
-		if (chain) return [{ name: node.id, dynamic }, ...chain];
+		if (chain) return [{ name: nodeId, dynamic }, ...chain];
 	}
 
 	for (const id of node.dynamicallyImportedIds) {
 		const child = node_getter(id);
 		const chain =
 			child && find_illegal_rollup_imports(node_getter, child, true, illegal_imports, seen);
-		if (chain) return [{ name: node.id, dynamic }, ...chain];
+		if (chain) return [{ name: nodeId, dynamic }, ...chain];
 	}
+
+	seen.delete(nodeId);
+	return null;
 };
 
 /**
  * Throw an error if a private module is imported from a client-side node.
  * @param {import('vite').ModuleNode} node
- * @param {Set<string>} illegal_imports
+ * @param {Set<string>} illegal_imports Illegal module IDs -- be sure to call vite.normalizePath!
  * @param {string} out_dir The directory specified by config.kit.outDir
  */
 export function prevent_illegal_vite_imports(node, illegal_imports, out_dir) {
 	const chain = find_illegal_vite_imports(node, illegal_imports);
 	if (chain) throw new Error(format_illegal_import_chain(chain, out_dir));
 }
 
+/*
+ * Vite does some weird things with import trees in dev
+ * for example, a Tailwind app.css will appear to import
+ * every file in the project. This isn't a problem for
+ * Rollup during build.
+ */
+const module_types = new Set([
+	'.ts',
+	'.js',
+	'.svelte',
+	'.mts',
+	'.mjs',
+	'.cts',
+	'.cjs',
+	'.svelte.md',
+	'.svx',
+	'.md'
+]);
```

Comment:
this hardcoded list makes me nervous. At the very least shouldn't it include `config.extensions` and `config.kit.moduleExtensions`?